helm/hydra/.rendered.yaml

```diff
@@ -0,0 +1,153 @@
+---
+# Source: hydra/templates/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/name: hydra
+    helm.sh/chart: hydra-0.0.1
+    app.kubernetes.io/managed-by: Tiller
+    app.kubernetes.io/instance: hydra
+  name: hydra
+spec:
+# Provides options for the service so chart users have the full choice
+  type: "ClusterIP"
+  ports:
+    - name: http
+      port: 80
+      protocol: TCP
+      targetPort: 4444
+  selector:
+    app.kubernetes.io/name: hydra
+    app.kubernetes.io/instance: hydra
+
+---
+# Source: hydra/templates/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hydra
+  labels:
+    app.kubernetes.io/name: hydra
+    helm.sh/chart: hydra-0.0.1
+    app.kubernetes.io/version: 0.0.1
+    app.kubernetes.io/instance: hydra
+    app.kubernetes.io/managed-by: Tiller
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: hydra
+      app.kubernetes.io/instance: hydra
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: hydra
+        app.kubernetes.io/instance: hydra
+    spec:
+      initContainers:
+        - name: hydra-migrate
+          image: "oryd/hydra:v1.0.0-rc.8_oryOS.10"
+          imagePullPolicy: IfNotPresent
+          command: ["hydra"]
+          args: [
+              "migrate",
+              "sql",
+              "postgres://$(DB_USERNAME):$(DB_PASSWORD)@$(DB_HOST):$(DB_PORT)/$(DB_NAME)"
+            ]
+          envFrom:
+            - configMapRef:
+                name: "hydra-env"
+          env:
+            - name: DB_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: hydra-db-secrets
+                  key: username
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: hydra-db-secrets
+                  key: password
+            - name: SYSTEM_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: hydra-secret
+                  key: system
+      containers:
+        - name: "hydra-public"
+          image: "oryd/hydra:v1.0.0-rc.8_oryOS.10"
+          imagePullPolicy: IfNotPresent
+          command: ["hydra"]
+          args: ["serve", "public"]
```

Comment:
Is there an upside to running those two commands separately? Keep in mind that it won't work with DSN=memory if split.

Reply:
This is the heritage of how we run Hydra — we have 3 _sets_ of containers running:
1. `hydra-admin` — 2 or 3 copies ran as `serve admin` for minimum fault-tolerance, only available inside the cluster
2. `hydra-intern` — N copies of `serve public` scaled to the needs of internal clients, only available inside the cluster
3. `hydra-public` — M copies of `serve public` scaled to the needs of the public clients, available on the edge to anyone who hits the `OAUTH2_ISSUER_URL`

Initial commit to this PR does not implement the scalability per-se, but it showcases our ability to provide Helm chart users with an option:
https://github.com/ory/k8s/blob/33d2d8b0681fd2f845b16a8d3a1d21069be5db66/helm/hydra/values.yaml#L23
which will result in manifest output you see `.rednered.yaml` if set to `true`, and if set to `false` will generate `PodSpec` with a *single container* running `hydra serve both`.

All of this is TBD.

Reply:
I think that's a great idea with the singleton - we should probably default to `true`.